packages/marko/src/core-tags/migrate/util/import-tag.js

```diff
@@ -7,19 +7,31 @@ module.exports = function importTag(baseImportPath, context) {
     identifiers: {},
     paths: {}
   });
+
+  // Create a camelcased tagName identifier based off of the matched file name.
+  // Replace any invalid JS chars (We don't need to worry about reserved words because of camelcase).
+  const createTagName = fileName =>
+    camelCase(fileName, {
+      pascalCase: true
+    }).replace(/^[^$A-Z_]|[^0-9A-Z_$]/gi, "_");
+
+  const getIdentifierName = importPath => {
+    let fileName;
+    if (importPath[0] === "<") fileName = importPath.slice(1, -1);
+    else {
+      const match = nameReg.exec(importPath);
+      fileName = (match && match[1]) || "template";
+    }
+    return createTagName(fileName);
+  };
+
   const importPath = context.getRelativePath(baseImportPath);
```

Comment:
I just realized: `importPath` on this line (used as a cache key below) would now be passing the shorthand such as `<component-a>` to the `getRelativePath` helper. I don't think we want that in this case, instead it should just use the `baseImportPath` as is.